gateway-ha/src/main/java/io/trino/gateway/ha/handler/QueryIdCachingProxyHandler.java

```diff
@@ -269,26 +273,38 @@ private boolean isPathWhiteListed(String path)
                 || extraWhitelistPaths.stream().anyMatch(s -> path.startsWith(s));
     }
 
-    public boolean handleAuthRequest(HttpServletRequest request)
+    @Override
+    public Optional<Cookie> removeCookie(HttpServletRequest clientRequest)
     {
-        return true;
+        if (removeCookiePaths.isEmpty() || removeCookiePaths.stream().noneMatch(c -> clientRequest.getRequestURI().startsWith(c))) {
+            return Optional.empty();
+        }
+
+        Optional<Cookie> cookie = Arrays.stream(clientRequest.getCookies()).filter(c -> c.getName().equalsIgnoreCase(COOKIE_NAME)).findAny();
+        clientRequest.getSession().invalidate();
+        cookie.ifPresent(c -> routingManager.removeCookie(sessionIdManager.getId(c.getValue())));
+
+        return cookie.map(c -> {
+            c.setMaxAge(0);
```

Comment:
can we have utility function to delete the cookies which we can reuse with the same values? UI also sets up some cookies.

Reply:
i updated the function to delete cookies based on the metadata set in the cookie. It will only delete the cookies it sets up itself - is that what you need?